apollo-router/src/plugins/connectors/make_requests.rs

```diff
@@ -18,18 +19,19 @@ const REPRESENTATIONS_VAR: &str = "representations";
 const ENTITIES: &str = "_entities";
 const TYPENAME: &str = "__typename";
 
-#[derive(Debug, Default)]
-struct RequestInputs {
+#[derive(Clone, Debug, Default)]
+pub(crate) struct RequestInputs {
```

Comment:
not a change for this PR, but because we're storing more stuff from various places for use in URL templates and selections, and we end up cloning all of that stuff into a map, i've been wondering:

should we change the URLTemplate::generate_path and Selection::apply_with_vars to take some kind of map-like trait that allows us to store Arc's of config, context, request headers, and other values without cloning?

Reply:
I thought the same thing—I'm going to add a public preview issue for generally auditing performance and doing a sweep of any easy clean-ups.